src/cleanup-text-tracks.js

```diff
@@ -26,6 +26,15 @@ export const removeExistingTrack = function(player, kind, label) {
  * @private
  */
 export const cleanupTextTracks = function(player) {
-  removeExistingTrack(player, 'captions', 'cc1');
+  const tracks = player.remoteTextTracks() || [];
+
+  for (let i = tracks.length - 1; i >= 0; i--) {
+    const track = tracks[i];
+
+    if (track.kind === 'captions' || track.kind === 'subtitles') {
```

Comment:
They are supposed to be (this comment made me realize I've been using the `addRemoteTextTrack` api wrong all this time as they aren't actually getting properly cleaned). The subtitle tracks for webvtt are taken care of in `MasterPlaylistController` of contrib-hls.